### PR TITLE
[1.11] getArmorModel is no longer Nullable

### PR DESCRIPTION
**Pull request by The-Fireplace**

### CODE REVIEW

File: patches/minecraft/net/minecraft/item/Item.java.patch [hunk 1]
```diff
@@ -386,10 +386,9 @@
 +     * @return  A ModelBiped to render instead of the default
 +     */
 +    @SideOnly(Side.CLIENT)
-+    @Nullable
 +    public net.minecraft.client.model.ModelBiped getArmorModel(EntityLivingBase entityLiving, ItemStack itemStack, EntityEquipmentSlot armorSlot, net.minecraft.client.model.ModelBiped _default)
```
Comment (liach) [inline patches/minecraft/net/minecraft/item/Item.java.patch:389]: Is the package `@ParametersAreNonnullByDefault`? If no then I suggest to add `@Nonnull` to the arguments also.

**Review by liach — COMMENTED** [review]
Users of this method should have their old null checks removed and existing overrides in the projects also should be annotated `@Nonnull`.

**Comment by liach** [comment]
@The-Fireplace One usage I found that could be simplified in [ForgeHookClients.java](https://github.com/MinecraftForge/MinecraftForge/blob/c177219ff3742c9dd9c383f99e2670635e28088f/src/main/java/net/minecraftforge/client/ForgeHooksClient.java#L203-L207):
```java
public static ModelBiped getArmorModel(EntityLivingBase entityLiving, ItemStack itemStack, EntityEquipmentSlot slot, ModelBiped _default)
{
    return itemStack.getItem().getArmorModel(entityLiving, itemStack, slot, _default);
}
```

File: patches/minecraft/net/minecraft/item/Item.java.patch (hunk 1 above)
Comment (The-Fireplace) [reply patches/minecraft/net/minecraft/item/Item.java.patch:389]: Yes, I believe the package is @ParametersAreNonnullByDefault

**Comment by The-Fireplace** [comment]
@liach good point, I'll get on it right away, and update the PR when I've done it.

**Comment by LexManos** [comment]
Nullability is fine, and changing the return value just because *SOME* ides give a warning will not fly.
Better to keep the existing functionality/contract and add the @Nullable to shutup some IDEs

**Comment by The-Fireplace** [comment]
Honestly, @LexManos , what initially drew my attention to the need to do this was that at one point(It has been recently fixed), it wasn't marked as Nullable, and the package was marked as ParametersAreNonnullByDefault, and it was easily possible to make it not return null.

At this point, however, I see no reason to have the method return null if there is a default value that can be returned instead, seeing as that default is the one that would be used.

**Comment by mezz** [comment]
This method was already marked correctly, but the new behavior makes it cleaner to use.

**Comment by The-Fireplace** [comment]
There, I couldn't find any other usages to simplify. So now I believe this is ready.

**Comment by LexManos** [comment]
I don't see a need for this, it changes functionlaiaty and could cause issuers on callers.
The cost/benifit is all off.

**Pull request closed without merging**